src/tests/unit/components/search-filter-input/SearchFilterInput.spec.jsx

```diff
@@ -0,0 +1,84 @@
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { useState } from 'react'
+import { beforeEach, describe, expect, vi } from 'vitest'
+import InputWithIcon from '~/components/input-with-icon/InputWithIcon'
+import SearchFilterInput from '~/components/search-filter-input/SearchFilterInput'
+
+const mockUpdateFilter = vi.fn()
+
+const defaultProps = {
+  updateFilter: mockUpdateFilter,
+  textFieldProps: { placeholder: 'Search...' }
+}
+
+describe('SearchFilterInput', () => {
+  let user
+
+  beforeEach(() => {
+    user = userEvent.setup()
+  })
+
+  it('should render component with input in it', () => {
+    render(<SearchFilterInput {...defaultProps} />)
+
+    expect(screen.getByRole('textbox')).toBeInTheDocument()
+    expect(screen.getByRole('button', { name: /search/i })).toBeInTheDocument()
+  })
+
+  it('should render typed text correctly', async () => {
+    render(<SearchFilterInput {...defaultProps} />)
+
+    const inputWithIcon = screen.getByRole('textbox')
+    await user.type(inputWithIcon, 'some text')
+
+    expect(inputWithIcon).toHaveValue('some text')
+  })
+
+  it('should delete typed text when delete button is clicked', async () => {
+    const Component = () => {
+      const [text, setText] = useState('Some text')
+      return (
+        <InputWithIcon
+          onClear={() => setText('')}
+          startIcon={<span>❌</span>}
+          value={text}
+        />
+      )
+    }
+
+    render(<Component />)
```

Comment:
Is it possible to mock `InputWithIcon` instead here? It seems like using `useState` and the custom component might be unnecessary and could complicate things a bit. Overall, good job!

Reply:
I've already simplified my test setup by mocking the InputWithIcon component.